lib/ansible/modules/clustering/k8s.py

```
@@ -0,0 +1,529 @@
+...
+'''
+
+class K8sException(Exception):
+    pass
+
+
+class K8sModule(AnsibleModule):
+
+    def __init__(self, argspec=None, mutually_exclusive=None, supports_check_mode=False):
+
+        if not HAS_K8S_MODULE_HELPER:
+            raise K8sException(
+                "This module requires the OpenShift Python client. Try `pip install openshift`"
+            )
+
+        if not HAS_YAML:
+            raise K8sException(
+                "This module requires PyYAML. Try `pip install PyYAML`"
+            )
+
+        AnsibleModule.__init__(self,
+                               argument_spec=argspec,
+                               supports_check_mode=supports_check_mode,
+                               mutually_exclusive=mutually_exclusive)
+
+        self.kind = self.params.pop('kind')
+        self.api_version = self.params.pop('api_version')
+        self.resource_definition = self.params.pop('resource_definition')
+        self.src = self.params.pop('src')
+        if self.src:
+            self.resource_definition = self.load_resource_definition(self.src)
+
+        if self.resource_definition:
+            self.api_version = self.resource_definition.get('apiVersion')
+            self.kind = self.resource_definition.get('kind')
+
+        self.api_version = self.api_version.lower()
+        self.kind = self._to_snake(self.kind)
+
+        try:
+            self.helper = self._get_helper(self.api_version, self.kind)
+        except Exception as exc:
+            raise K8sException(
+                "Error initializing AnsibleModuleHelper: {}".format(str(exc))
+            )
+
+    @staticmethod
+    def _get_helper(api_version, kind):
+        try:
+            helper = KubernetesAnsibleModuleHelper(api_version, kind)
+            helper.get_model(api_version, kind)
+        except:
+            try:
+                helper = OpenShiftAnsibleModuleHelper(api_version, kind)
+                helper.get_model(api_version, kind)
+            except:
```

Just noting the second bare except